Falcon/Falcon/Rendering/PipeLine/Renderer.cpp

```diff
@@ -146,48 +153,52 @@ void Renderer::SetDrawStates(boost::container::vector<Entity*, fmemory::StackSTL
 */
 
 float temp = 0.0f;
-void Renderer::Update(int width, int height, Camera &cam, float dt, boost::container::vector<Entity*, fmemory::StackSTLAllocator<Entity*>>* entities)
+void Renderer::Update(Camera &cam, float dt, boost::container::vector<Entity*, fmemory::StackSTLAllocator<Entity*>>* entities)
 {
 	temp += 1.0f * dt;
 	m_RES->ProcessEvents();
-	glm::mat4 projection = glm::perspective(glm::radians(cam.m_Zoom), (float)width / (float)height, 0.1f, 100.0f);
-
-	shader->UseShader();
-	shader->SetMat4("projection", projection);
-	shader->SetMat4("view", cam.GetViewMatrix());
+	m_entities = entities;
+	for (unsigned int i = 0; i < m_entities->size(); ++i)
+	{
+		Shader* shader = m_entities->at(i)->GetComponent<RenderComponent>()->m_mesh->GetMaterial()->m_shader;
```

Comment:
Not all entities will have the rendercomponent, so you should check if it does before accessing it's variables

Reply:
yeah I wondered why it wasn't crashing on visual studios without that check. Seems visual studio is effectively ruining programming.